### ISSUE TITLE
wall model in Modelica.Fluid.Examples.HeatExchanger.BaseClasses.BasicHX

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1352]
**Reported by ulf&#46;nordstrom on 15 Nov 2013 13:40 UTC**
Hi,
One of our users reported

"
The "wall" model Modelica.Fluid.Examples.HeatExchanger.BaseClasses.BasicHX references the average of the pipes cross sectional areas for the heat transfer area area_h. Shouldn't it reference the pipe parameter area_h_1 and area_h_2? 
"

Could you have a look at it.

Best regards,
Ulf

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1352

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 21 May 2014 20:41 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 26 Jun 2014 15:45 UTC**
I assume the proposal to change in Modelica.Fluid.Examples.HeatExchanger.BaseClasses.BasicHX:
```
  BaseClasses.WallConstProps wall(
    ...
    area_h=(crossArea_1 + crossArea_2)/2)
```
to
```
  BaseClasses.WallConstProps wall(
    ...
    area_h=area_h)
```
I think you are right, but someone from the Fluid group should verify

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 26 Jun 2014 18:07 UTC**
Martin, I think your assumption is right. With the current settings heat transfer is about 2 orders of magnitude too low. 

BTW: the example ha a number of other problems. This error would have been detected easily, if customary variables for heat exchangers like the heatcapacity flow would have been included. 

The model shows how you _can_ build a heat exchanger from the base classes in Modelica.Fluid, but from the point of view of usability it is atrocious, seriously very bad. I am also not sure what the example is supposed to show. 

I leave it to active members of the fluid group what to do about this, but it seems to me rather a "how not to use object orientation" counter-example.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 27 Jun 2014 07:36 UTC**
Fixed in 1347a2db16a852a7a5f583239c3be93145374df7 (trunk) and in 3614de4d4bc8c5145a688de4854246c1d4cefe89 (maintenance)
